lib/net.js

```diff
@@ -271,40 +280,50 @@ function Socket(options) {
   if (options.handle) {
     this._handle = options.handle; // private
     this[async_id_symbol] = getNewAsyncId(this._handle);
-  } else if (options.fd !== undefined) {
-    const { fd } = options;
-    let err;
-
-    // createHandle will throw ERR_INVALID_FD_TYPE if `fd` is not
-    // a valid `PIPE` or `TCP` descriptor
-    this._handle = createHandle(fd, false);
-
-    err = this._handle.open(fd);
+  } else {
+    const onread = options.onread;
+    if (onread !== null && typeof onread === 'object' &&
+        isUint8Array(onread.buffer) &&
+        typeof onread.callback === 'function') {
+      this[kBuffer] = onread.buffer;
+      this[kBufferCb] = onread.callback;
```

Comment:
From reading the documentation I would expect that `this` is `onread` in `onread.callback()` but it isn't.

Reply:
I'd probably rather just document that `this` is the global object/context? Would that suffice?

Reply:
Updated.